### PR TITLE
Read subscriptions from discovery behind feature flag

### PR DESCRIPTION
**Pull request by michellebrier**
### Description
Read subscriptions from discovery if the `READ_SUBSCRIBERS_FROM_DISCOVERY_ENABLED` flag is on.

### Dragons
N/A

### How Has This Been Tested?
- Turn flag on with breakpoint
- Go to a user you are subscribed to. Make sure discovery endpoint is hit, not the identity endpoint. Make sure the subscribe bell is toggled
- Go to a user you are not subscribed to. Make sure discovery endpoint is hit, not the identity endpoint. Make sure the subscribe bell is not toggled
- Repeat with flag off

### Feature Flags ###
`READ_SUBSCRIBERS_FROM_DISCOVERY_ENABLED`

### CODE REVIEW

File: packages/common/src/services/audius-backend/AudiusBackend.ts [hunk 1]
```diff
@@ -2775,29 +2777,56 @@ export const audiusBackend = ({
...
-    try {
-      const { data, signature } = await signData()
-      return await fetch(
-        `${identityServiceUrl}/notifications/subscription?userId=${userId}`,
-        {
-          headers: {
-            [AuthHeaders.Message]: data,
-            [AuthHeaders.Signature]: signature
+
+    const readSubscribersFromDiscoveryEnabled =
```
Comment (dylanjeffers) [inline packages/common/src/services/audius-backend/AudiusBackend.ts:2789]: nit, but we try to prefix booleans with `is`

File: packages/common/src/services/audius-backend/AudiusBackend.ts [hunk 2]
```diff
@@ -2775,29 +2777,56 @@ export const audiusBackend = ({
...
+        FeatureFlags.READ_SUBSCRIBERS_FROM_DISCOVERY_ENABLED
+      )) ?? false
+
+    if (readSubscribersFromDiscoveryEnabled) {
+      // Read subscribers from discovery
+      try {
+        const subscribers = await audiusLibs.User.getUserSubscribers(
+          encodeHashId(userId)
+        )
+        const subscriberIds = subscribers.map((s: APIUser) =>
```
Comment (dylanjeffers) [inline packages/common/src/services/audius-backend/AudiusBackend.ts:2800]: personal nit, i like still doing `subscriber` for the map argument, what you think?

File: packages/common/src/services/audius-backend/AudiusBackend.ts [hunk 3]
```diff
@@ -2775,29 +2777,56 @@ export const audiusBackend = ({
...
+
+    const readSubscribersFromDiscoveryEnabled =
+      (await getFeatureEnabled(
+        FeatureFlags.READ_SUBSCRIBERS_FROM_DISCOVERY_ENABLED
+      )) ?? false
+
+    if (readSubscribersFromDiscoveryEnabled) {
+      // Read subscribers from discovery
+      try {
+        const subscribers = await audiusLibs.User.getUserSubscribers(
```
Comment (dylanjeffers) [inline packages/common/src/services/audius-backend/AudiusBackend.ts:2797]: is there any way `getUserSubscribers` could return `ApiUser[]` so we don't have to type subscriber in the `.map`?

File: packages/common/src/services/audius-backend/AudiusBackend.ts [hunk 4]
```diff
@@ -2775,29 +2777,56 @@ export const audiusBackend = ({
     }
   }
 
+  /**
+   * Returns whether the current user is subscribed to userId.
+   * @returns Boolean
```
Comment (dylanjeffers) [inline packages/common/src/services/audius-backend/AudiusBackend.ts:2782]: with typescript, the @returns js-doc is not needed, we can either rely on the implicit return type, or you can explicitly type it with 
```
async function getUserSubscribed(userId: ID): Promise<boolean> {
```

**Review by dylanjeffers — APPROVED** [review]
nicee

File: packages/common/src/services/audius-backend/AudiusBackend.ts (hunk 1 above)
Comment (michellebrier) [reply packages/common/src/services/audius-backend/AudiusBackend.ts]: added in 37d3ffce99d3ea6ae2137e569042e69810a15b64

File: packages/common/src/services/audius-backend/AudiusBackend.ts (hunk 2 above)
Comment (michellebrier) [reply packages/common/src/services/audius-backend/AudiusBackend.ts]: sure, added in 37d3ffce99d3ea6ae2137e569042e69810a15b64

File: packages/common/src/services/audius-backend/AudiusBackend.ts (hunk 3 above)
Comment (michellebrier) [reply packages/common/src/services/audius-backend/AudiusBackend.ts]: added in 37d3ffce99d3ea6ae2137e569042e69810a15b64

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mbrier/plat-456/read-subscriptions-from-discprov

**Pull request merged**